packages/server/src/scripts/consumeGrantModifications.js

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env node
+const { getSQSClient } = require('../lib/gost-aws');
+const knex = require('../db/connection');
+const { processMessages, receiveNextMessageBatch } = require('../lib/grants-ingest');
+
+async function main() {
+    let shutDownRequested = false;
+    const requestShutdown = (signal) => {
+        console.log(`${signal} received. Requesting shutdown...`);
+        shutDownRequested = true;
+    };
+    process.on('SIGTERM', requestShutdown);
+    process.on('SIGINT', requestShutdown);
+
+    const queueUrl = process.env.GRANTS_INGEST_EVENTS_QUEUE_URL;
+    const sqs = getSQSClient();
+    while (shutDownRequested === false) {
+        console.log(`Long-polling next SQS message batch from ${queueUrl}`);
+        // eslint-disable-next-line no-await-in-loop
+        const messages = await receiveNextMessageBatch(sqs, queueUrl);
+        if (messages.length > 0) {
+            // eslint-disable-next-line no-await-in-loop
+            await processMessages(knex, sqs, queueUrl, messages);
+        }
```

Comment:
do we want to implement some type of waiting mechanism here? or is it fine to call `receiveNextMessageBatch` immediately after it returns an empty result?

Reply:
I'm concerned about a situation where the queue is returning some type of error and we keep polling.

Reply:
> do we want to implement some type of waiting mechanism here? or is it fine to call `receiveNextMessageBatch` immediately after it returns an empty result?

The SQS `ReceiveMessages` behavior underlying this call already implements long-polling, which will wait for up to 20 seconds before returning, so we always wait that long in scenarios where there aren't any messages available.

> I'm concerned about a situation where the queue is returning some type of error and we keep polling.

If we encounter an error from SQS when performing the `ReceiveMessages` action, the error is thrown and will cause the running process to crash, which is intentional for this very reason – in the event that there's a problem with SQS (unlikely, but possible) or how we're calling SQS (more likely; could be due to missing permissions or programming error, etc.), we want the ECS task to register an error that the container failed so that it can back off accordingly, rather than just continuing to fail forever.

A few additional pieces of context that may be worth noting:
1. In the case of an intermittent failure, where SQS is unreachable for a brief period and then comes back, a new ECS task execution will simply replace the one that crashed. Therefore, it's better for this script to "fail hard" and allow the system to react accordingly, either by self-healing automatically (by replacing the task that failed) or backing off entirely if all tasks are failing (which indicates that developer intervention/investigation is necessary).
2. The auto-scaling behavior that will be implemented by #1296 will add/remove ECS tasks running this script according to the number of messages in the SQS queue. Theoretically, the scenario in which this script starts to receive nothing but empty message batches would indicate that the queue size is getting low, which means that ECS will start shutting down (some of) the running ECS tasks as part of the configured "scale-down" behavior.